staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go

```diff
@@ -1039,6 +1040,145 @@ spec:
 	}
 }
 
+func TestNoTrackManagedFieldsForClientSideApply(t *testing.T) {
+	f := NewDefaultTestFieldManager(schema.FromAPIVersionAndKind("apps/v1", "Deployment"))
+
+	// create object
+	newObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	deployment := []byte(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment
+  labels:
+    app: my-app
+spec:
+  replicas: 100
+`)
+	if err := yaml.Unmarshal(deployment, &newObj.Object); err != nil {
+		t.Errorf("error decoding YAML: %v", err)
+	}
+	if err := f.Update(newObj, "test_kubectl_create"); err != nil {
+		t.Errorf("failed to update object: %v", err)
+	}
+	if m := f.ManagedFields(); len(m) == 0 {
+		t.Errorf("expected to have managed fields, but got: %v", m)
+	}
+
+	// stop tracking managed fields
+	newObj = &unstructured.Unstructured{Object: map[string]interface{}{}}
+	deployment = []byte(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment
+  managedFields: [] # stop tracking managed fields
```

Comment:
interesting, does this actually stop the tracking?

Reply:
Yes, confirmed also in this integration test:

https://github.com/kubernetes/kubernetes/blob/f2deb2417a6c542c54606ab17376b26ef1552b87/test/integration/apiserver/apply/apply_test.go#L1583-L1665

Reply:
Just a note that it's possible to only clear/stop tracking managed fields like this on an update, but not on a create.